### DISCUSSION TITLE
R4phd - R package installation

### DISCUSSION THREAD

**Discussion by giscus** [sorenoneill/r4phd#15] [General]
# R4phd - R package installation

How to install an R package on your computer

https://sorenoneill.github.io/r4phd/qmd_installation/installation_r_packages_en.html

**Comment by BibiHeiberg** [comment]
Hej Sten og Søren
Som måske ventet kan jeg ikke få lov at installere RTools på min SDU pc. Når jeg forsøger at installere beder pc om kontakt til administrators. Skal jeg gøre det eller er der et smart short cut?
Forsøgte at drible videre i installation af packages uden RTools, men så får jeg følgende fejlmelding (ja jeg prøvede tre gange for en sikkerheds skyld ;):

R version 4.3.2 (2023-10-31 ucrt) -- "Eye Holes"
Copyright (C) 2023 The R Foundation for Statistical Computing
Platform: x86_64-w64-mingw32/x64 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> source("~/.active-rstudio-document")
[1] "[OK] Your operating system is Windows (version 10 x64)"
[1] "[OK] You computer has a 64 bit CPU"
[1] "[OK] R installation is: R version 4.3.2 (2023-10-31 ucrt)"
Error in loadNamespace(x) : there is no package called ‘rstudioapi’
> source("~/.active-rstudio-document")
[1] "[OK] Your operating system is Windows (version 10 x64)"
[1] "[OK] You computer has a 64 bit CPU"
[1] "[OK] R installation is: R version 4.3.2 (2023-10-31 ucrt)"
Error in loadNamespace(x) : there is no package called ‘rstudioapi’
> source("~/.active-rstudio-document")
[1] "[OK] Your operating system is Windows (version 10 x64)"
[1] "[OK] You computer has a 64 bit CPU"
[1] "[OK] R installation is: R version 4.3.2 (2023-10-31 ucrt)"
Error in loadNamespace(x) : there is no package called ‘rstudioapi’

**Reply by sorenoneill** [reply]
Sten er mere rutineret i Windows end jeg -- han er varskoet og kigger på det a.s.a.p. :-)

**Reply by BibiHeiberg** [reply]
Perfekt -  det er lykkedes mig at installere RTools (tror jeg nok) - jeg har i hvert fald fået lov at downloade nu. Det krævede midlertidige admin rettigheder. Men jeg får fortsat samme fejlmelding når jeg tjekker om om "alt er ok".
> source("~/.active-rstudio-document")
[1] "[OK] Your operating system is Windows (version 10 x64)"
[1] "[OK] You computer has a 64 bit CPU"
[1] "[OK] R installation is: R version 4.3.2 (2023-10-31 ucrt)"
Error in loadNamespace(x) : there is no package called ‘rstudioapi’

**Reply by sorenoneill** [reply]
Det ser ud som om pakken 'rstudioapi' ikke er installeret -- jeg skal lige undersøge det nærmere, men foreløbigt, så prøv at køre flg kommando i RStudios konsol. Hvis du er i tvivl om hvor den er så kig her: https://sorenoneill.github.io/r4phd/qmd_gui/rstudio_introduction_to_gui_1_en.html

`install.packages("rstudioapi")`

Du skal have internet adgang for at downloade pakken.

**Comment by BibiHeiberg** [comment]
Det hjalp noget, men nu melder den så disse fejl:

[1] "[OK] Your operating system is Windows (version 10 x64)"
[1] "[OK] You computer has a 64 bit CPU"
[1] "[OK] R installation is: R version 4.3.2 (2023-10-31 ucrt)"
[1] "[OK] The RStudio installation is 'desktop'"
[1] "[OK] The RStudio version is 2023.12.0.369"
[1] "[WARNING] 'tidyverse' (package) is NOT installed."
[1] "[WARNING] 'here' (package) is NOT installed."
[1] "[WARNING] 'skimr' (package) is NOT installed."
[1] "[WARNING] 'gt' (package) is NOT installed."
[1] "[WARNING] 'gtExtras' (package) is NOT installed."
[1] "[WARNING] 'gtsummary' (package) is NOT installed."
[1] "[WARNING] 'patchwork' (package) is NOT installed."
[1] "[WARNING] 'naniar' (package) is NOT installed."
[1] "[WARNING] 'reprex' (package) is NOT installed."
[1] "[WARNING] 'ggstatsplot' (package) is NOT installed."
[1] "[WARNING] 'easystats' (package) is NOT installed."
[1] "[WARNING] 'vroom' (package) is NOT installed."

Er alle de pakker der mangler en del af Rtools42?

Venlig hilsen
Bibi Dige Heiberg
Ph.d.-stipendiat
Det Sundhedsvidenskabelige Fakultet
Institut for Idræt og Biomekanik
Forskningsenheden Klinisk Biomekanik
T       65 50 95 41<tel:+4565509541>
***@***.******@***.***>
portal.findresearcher.sdu.dk/da/persons/bheiberg<https://portal.findresearcher.sdu.dk/da/persons/bheiberg>
Syddansk Universitet
Campusvej 55
5230 Odense M
www.sdu.dk<https://www.sdu.dk/>
[https://cdn.sdu.dk/img/sdulogos/SDU_BLACK_signatur.png]
________________________________
Fra: sorenoneill ***@***.***>
Sendt: 8. januar 2024 18:56
Til: sorenoneill/r4phd ***@***.***>
Cc: Bibi Dige Heiberg ***@***.***>; Comment ***@***.***>
Emne: Re: [sorenoneill/r4phd] R4phd - R package installation (Discussion #15)

Du får ikke ofte mails fra ***@***.*** Få mere at vide om, hvorfor dette er vigtigt<https://aka.ms/LearnAboutSenderIdentification>

Det ser ud som om pakken 'rstudioapi' ikke er installeret -- jeg skal lige undersøge det nærmere, men foreløbigt, så prøv at køre flg kommando i RStudios konsol. Hvis du er i tvivl om hvor den er så kig her: https://sorenoneill.github.io/r4phd/qmd_gui/rstudio_introduction_to_gui_1_en.html

install.packages("rstudioapi")

Du skal have internet adgang for at downloade pakken.

**Comment by BibiHeiberg** [comment]
Hej igen

Jeg kan se at Rtools42 er hentet og lægger på c-drevet, men når jeg forsøger at åbne får jeg følgende fejlmelding:

[cid:18c82734-70a4-439d-aebe-04c4117ebd0a]

Venlig hilsen
Bibi Dige Heiberg
Ph.d.-stipendiat
Det Sundhedsvidenskabelige Fakultet
Institut for Idræt og Biomekanik
Forskningsenheden Klinisk Biomekanik
T       65 50 95 41<tel:+4565509541>
***@***.******@***.***>
portal.findresearcher.sdu.dk/da/persons/bheiberg<https://portal.findresearcher.sdu.dk/da/persons/bheiberg>
Syddansk Universitet
Campusvej 55
5230 Odense M
www.sdu.dk<https://www.sdu.dk/>
[https://cdn.sdu.dk/img/sdulogos/SDU_BLACK_signatur.png]
________________________________
Fra: sorenoneill ***@***.***>
Sendt: 8. januar 2024 18:56
Til: sorenoneill/r4phd ***@***.***>
Cc: Bibi Dige Heiberg ***@***.***>; Comment ***@***.***>
Emne: Re: [sorenoneill/r4phd] R4phd - R package installation (Discussion #15)

Du får ikke ofte mails fra ***@***.*** Få mere at vide om, hvorfor dette er vigtigt<https://aka.ms/LearnAboutSenderIdentification>

Det ser ud som om pakken 'rstudioapi' ikke er installeret -- jeg skal lige undersøge det nærmere, men foreløbigt, så prøv at køre flg kommando i RStudios konsol. Hvis du er i tvivl om hvor den er så kig her: https://sorenoneill.github.io/r4phd/qmd_gui/rstudio_introduction_to_gui_1_en.html

install.packages("rstudioapi")

Du skal have internet adgang for at downloade pakken.

**Reply by sorenoneill** [reply]
Listen herover er alle pakker du ikke har fået installeret endnu.

Jeg bemærker at du har installeret R version 4.3 og RTools 4.2 -- jeg har som sagt ikke selv Windows, men R og Rtools skal være samme versions nummer (på de første to cifre)
